sklearn/preprocessing/_data.py

```diff
@@ -2262,6 +2262,9 @@ def _dense_fit(self, X, random_state):
                 col = col.take(subsample_idx, mode='clip')
             self.quantiles_.append(np.nanpercentile(col, references))
         self.quantiles_ = np.transpose(self.quantiles_)
+        # due to floating-point precision error in `np.nanpercentile`,
+        # make sure that quantiles are monotonically increasing
+        self.quantiles_ = np.maximum.accumulate(self.quantiles_)
```

Comment:
In addition to the non-regression test, please improve the inline comment to add a link the upstream issue in numpy:
`https://github.com/numpy/numpy/issues/14685`